projects/wallet-template/src/background/keyring.ts

```diff
@@ -4,36 +4,150 @@ import {
   ed25519CreateDerive,
   sr25519CreateDerive,
 } from "@polkadot-labs/hdkd"
-import { keystoreV4, type KeystoreV4WithMeta } from "./keystore"
+import { sr25519, ed25519, ecdsa, KeyPair } from "@polkadot-labs/hdkd-helpers"
+import { KeystoreMeta, keystoreV4, type KeystoreV4WithMeta } from "./keystore"
 import { assert } from "./utils"
 import * as storage from "./storage"
-import { Keyset } from "./types"
+import { InsertKeysetArgs, KeysetAccount } from "./types"
+import { fromHex, toHex } from "@polkadot-api/utils"
 
 const createDeriveFnMap = {
   Sr25519: sr25519CreateDerive,
   Ed25519: ed25519CreateDerive,
   Ecdsa: ecdsaCreateDerive,
 } as Record<string, CreateDeriveFn>
 
+const privateKeyToPublicKey = (privateKey: string, scheme: string) => {
+  switch (scheme) {
+    case "Sr25519":
+      return toHex(sr25519.getPublicKey(privateKey))
+    case "Ed25519":
+      return toHex(ed25519.getPublicKey(privateKey))
+    case "Ecdsa":
+      return toHex(ecdsa.getPublicKey(privateKey))
+    default:
+      throw new Error("unsupported scheme")
+  }
+}
+
 export const createKeyring = () => {
   const getKeystore = () => storage.get("keystore")
   const setKeystore = (keystore: KeystoreV4WithMeta) =>
     storage.set("keystore", keystore)
   const removeKeystore = () => storage.remove("keystore")
-  const getKeysets = async () =>
-    (await getKeystore())?.meta.filter((m) => m.type === "keyset") ?? []
+
+  const getKeyStoreAccounts = (keystoreMeta: KeystoreMeta): KeysetAccount[] => {
+    switch (keystoreMeta._type) {
+      case "DerivationPathKeystore":
+        return keystoreMeta.derivationPaths.map((d) => ({
+          ...d,
+          _type: "DerivationPath",
+        }))
+      case "PrivateKeyKeystore":
+        return [
+          {
+            _type: "Keypair",
+            publicKey: privateKeyToPublicKey(
+              keystoreMeta.privateKey,
```

Comment:
`privateKey` should be encrypted in the secrets and not unencrypted in the meta

<img width="390" alt="image" src="https://github.com/paritytech/substrate-connect/assets/1209171/103afb19-8e7d-4c30-a8e7-4ce5dd05936b">